platform/linux/mod.rs

```diff
@@ -278,16 +286,10 @@ impl UnixSender {
 
                 if result <= 0 {
                     let error = UnixError::last();
-                    if error.0 == libc::ENOBUFS && bytes_to_send > 2000 {
+                    if error.0 == libc::ENOBUFS
+                       && downsize(&mut sendbuf_size, bytes_to_send).is_ok() {
```

Comment:
I don't think returning the empty error from `downsize()` would be correct handling... The idea here is that if the downsize fails, we want to return the original error -- which happens in the else case.

As I said in the commit message, I'm not entirely happy how this turned out syntactically -- but I really can't think of a better approach. Quite frankly, I considered dropping this size test entirely: it's a case that I don't actually expect to happen; I only added it as an additional safeguard in the original `ENOBUFS` handling because it seemed easy to do -- but it turned out to be a major pain during refactoring, and I'm not sure any more it's indeed worth the trouble...

Maybe we should just turn it into an `assert()` instead?